src/browser/modules/Editor/EditorFrame.tsx

```diff
@@ -79,46 +79,80 @@ import {
   createFilePath
 } from 'browser/modules/Sidebar/project-files.utils'
 import { defaultFavoriteName } from 'browser/modules/Sidebar/favorites.utils'
+import Monaco, { MonacoHandles } from './Monaco'
+import {
+  getTheme,
+  LIGHT_THEME,
+  shouldEnableMultiStatementMode
+} from 'shared/modules/settings/settingsDuck'
+import useDerivedTheme from 'browser-hooks/useDerivedTheme'
+import { BrowserTheme } from './CypherMonacoThemes'
+import * as schemaConvert from './editorSchemaConverter'
+import cypherFunctions from './cypher/functions'
+import { getUseDb } from 'shared/modules/connections/connectionsDuck'
+import { getHistory, HistoryState } from 'shared/modules/history/historyDuck'
+
+interface EditorSupportSchema {
+  labels?: string[]
+  relationshipTypes?: string[]
+  propertyKeys?: string[]
+  functions?: string[]
+  procedures?: string[]
+  consoleCommands: string[]
+  parameters?: string[]
+}
 
 type EditorFrameProps = {
+  browserTheme: BrowserTheme
   bus: Bus
-  theme: { linkHover: string }
+  enableMultiStatementMode: boolean
   executeCommand: (cmd: string, source: string) => void
-  updateFavorite: (id: string, value: string) => void
+  history: HistoryState
   projectId: string
-}
-type CodeEditor = {
-  getValue: () => string | null
-  setValue: (newText: string) => void
+  schema: EditorSupportSchema
+  theme: { linkHover: string }
+  updateFavorite: (id: string, value: string) => void
+  useDb: null | string
 }
 
 type SavedScript = {
-  id: string
   content: string
+  directory?: string
+  id: string
   isProjectFile: boolean
   isStatic: boolean
   name?: string
-  directory?: string
 }
 
 export function EditorFrame({
+  browserTheme,
   bus,
-  theme,
+  enableMultiStatementMode,
   executeCommand,
+  history,
+  projectId,
+  schema,
+  theme,
   updateFavorite,
-  projectId
+  useDb
 }: EditorFrameProps): JSX.Element {
   const [addFile] = useMutation(ADD_PROJECT_FILE)
   const [unsaved, setUnsaved] = useState(false)
   const [isFullscreen, setFullscreen] = useState(false)
   const [currentlyEditing, setCurrentlyEditing] = useState<SavedScript | null>(
     null
   )
-  const editorRef = useRef<CodeEditor>(null)
+  const editorRef = useRef<MonacoHandles>(null)
+  const [lineCount, setLineCount] = useState(1)
```

Comment:
I can't see what the lineCount is used for?

Reply:
Apart from rerenders

Reply:
This is a leftover that is not needed anymore, removed